### DISCUSSION TITLE
Alt Text / Image Descriptions

### DISCUSSION THREAD

**Discussion by Robo-Fortune** [stoatchat/discussions#584] [Feature Suggestions]
### Reason for Request
Many users benefit from the use of alt-text / descriptions on images in platforms with text size limits. While Revolt doesn't have such limits and thus regular image descriptions can be written out, on preformatted sections of text (such as ASCII Art) that aren't otherwise accessible to users of screen readers or refreshable braille displays, an alt-text feature could come in handy. Such ASCII art is common on Discord, and as this platform broadly competes with it and draws users from it, it's likely that ASCII art will become common here as well (especially given the presence of good markdown support, enabling preformatted text to be easily written, even included in bios). For this reason, it would likely be helpful (and indeed, situationally necessary for many users) to have an alt-text feature built in for pre-formatted text blocks.

### Proposed Functionality
When creating a pre-formatted text block (I:E three back-ticks on either side of text), there should be a small modal that appears to prompt users for whether they'd like to add alt-text which describes the section. Screen reader programs or similar applications should be instructed to ignore the contents of any block text element that has alt-text associated with it, and to instead read out the alt text. As for how specifically to implement this, I've thus far been unable to find any documentation on how screen readers are handled by Revolt in general, so I cannot give specifics. However, consideration for screen readers has clearly been given to the web client, which includes specific instructions on how to move elements when using a screen reader, so it's evident that this is broadly a user-base that Revolt aims to support, and preventing ascii art from bombarding them with a stream of "w"s or "{i"s would likely be appreciated.

**Comment by alisticks** [comment]
i would also really love this! it would also be good for screenshots since revolt has some hard-to-see themes. accessibility is important!
